### DISCUSSION TITLE
Completely new API...again

### DISCUSSION THREAD

**Discussion by jonathansgardner** [TanStack/table#5682] [General]
Completey new library yet again. This is the second time I've had to update this package in a project I maintain and it has a completely new API and every single thing about it has changed. The migration guide just has only a few basic examples. Will it always be completely breaking from version to version?

**Comment by KevinVandy** [comment]
Are you talking about v8 from a couple years ago or the new v9 which is still in alpha?

**Reply by jonathansgardner** [reply]
v8 from a couple years ago. Let me guess, v9 changes the entire API again?

**Reply by jonathansgardner** [reply]
Also it's been out for a couple years and the migration guide is still that bad? I thought that the docs for v7 were so good. It was a pleasure to put the components together. Now I just have to scrap that entire setup and start from scratch because I need to use some of the features that don't exist in v7.

**Reply by KevinVandy** [reply]
You have the option of only migrating 1 table at a time to V8 as it lives in a new package name (`react-table` vs `@tanstack/react-table`. TanStack Table V8 was a complete rewrite of the library in TypeScript. Many of the new capabilities are because of the new rewrite. If using TanStack Table makes your life easier, use it. If it doesn't, don't.

Major version bumps have breaking changes. This is expected. They wouldn't be major version bumps otherwise.

V9 will be a much smaller upgrade than V8 was. The goal is that you will not have to change any of the actual markup logic. The changes will all be confined to how the `useTable` hook is defined.

**Comment by jonathansgardner** [comment]
Well anyway, I don't have that option as it's a project I'm maintaining that needs to be updated. It's not just a few breaking changes though. Every single value that is imported or returned by the useReactTable hook has changed in some way. And there is no way the markup or logic stays the same either. Thanks for your answer.
